app/src/ui/preferences/preferences.tsx

```diff
@@ -110,16 +110,16 @@ export class Preferences extends React.Component<
     const availableEditors = editors.map(e => e.editor)
     const availableShells = shells.map(e => e.shell)
 
-    this.setState({
+    this.setState((_state, props) => ({
       committerName,
```

Comment:
This is also erroring because we moved `committerName` and `committerEmail` into a different scope, so TSC doesn't trust that it's non-null. Here's a fix that works for me:

```diff
@@ -98,9 +98,6 @@ export class Preferences extends React.Component<
       }
     }
 
-    committerName = committerName || ''
-    committerEmail = committerEmail || ''
-
     const [editors, shells, mergeTool] = await Promise.all([
       getAvailableEditors(),
       getAvailableShells(),
@@ -110,16 +107,16 @@ export class Preferences extends React.Component<
     const availableEditors = editors.map(e => e.editor)
     const availableShells = shells.map(e => e.shell)
 
-    this.setState({
-      committerName,
-      committerEmail,
-      optOutOfUsageTracking: this.props.optOutOfUsageTracking,
-      confirmRepositoryRemoval: this.props.confirmRepositoryRemoval,
-      confirmDiscardChanges: this.props.confirmDiscardChanges,
+    this.setState((_prevState, props) => ({
+      committerName: committerName || '',
+      committerEmail: committerEmail || '',
+      optOutOfUsageTracking: props.optOutOfUsageTracking,
+      confirmRepositoryRemoval: props.confirmRepositoryRemoval,
+      confirmDiscardChanges: props.confirmDiscardChanges,
       availableShells,
       availableEditors,
       mergeTool,
-    })
+    }))
   }
 
   public render() {
```